### PR TITLE
Upgrade webpack to latest

### PR DESCRIPTION
**Pull request by Hyzual**
webpack's (major) changelog: https://github.com/webpack/webpack/releases/tag/v4.0.0
file-loader (major) changelog: https://github.com/webpack-contrib/file-loader/blob/master/CHANGELOG.md

Paths to images have been changed to relative paths. Previous paths
resulted in errors like "Cannot find module 'null'".

I have not upgraded sass-loader and vue-loader yet, let's do this in a
dedicated contribution.

How to test:
- You will need to checkout the pull request for koel (server) to have
  the matching laravel-mix version.
- yarn run production (at koel server root) should succeed and build the
  assets

Note to reviewer:
- I removed node 8 from targets of travis CI. file-loader v5.0.0 requires nodejs >= 10.13.0 [see changelog](https://github.com/webpack-contrib/file-loader/blob/master/CHANGELOG.md#500-2019-11-22)
- I have updated snapshots but it looks like some changes were not due to this contribution.
For example `js/tests/components/layout/main-wrapper/index.spec.js`. I tried to stub manually the "main-content.vue" component because it kept changing (greetings are randomized) and breaking the snapshot.

### CODE REVIEW

File: js/tests/components/layout/main-wrapper/index.spec.js [hunk 1]
```diff
@@ -1,5 +1,12 @@
 import Component from '@/components/layout/main-wrapper/index'
 
+// Mock manually this component because it changes randomly and breaks the snapshot
+jest.mock('@/components/layout/main-wrapper/main-content', () => {
```
Comment (phanan) [inline js/tests/components/layout/main-wrapper/index.spec.js:4]: Do you think we can mock the `greeting` computed property instead?

Comment (Hyzual) [reply js/tests/components/layout/main-wrapper/index.spec.js:4]: I'm not sure how to do that. It's a dependency's dependency's computed prop. MainContent is even dynamically imported, which I guess is why `@vue/test-utils` is not stubbing it despite the `shallowMount()`.

It looks like `@vue/test-utils` stubs dynamic imported components in [v1.0.0-beta.27](https://github.com/vuejs/vue-test-utils/blob/dev/CHANGELOG.md#100-beta27-2018-12-09)
Upgrading it would let us remove this manual mock and have a snapshot that does not change all the time. What do you think about it ?

Comment (Hyzual) [reply js/tests/components/layout/main-wrapper/index.spec.js:4]: Of course when I say "let's upgrade `@vue/test-utils`", I mean I'll try and submit another PR to keep this one small and easy to review.

File: js/components/meta/about-dialog.vue [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
     <main>
       <div class="logo">
-        <img src="~#/../img/logo.svg" width="128" height="auto">
+        <img src="../../../img/logo.svg" width="128" height="auto">
```
Comment (phanan) [inline js/components/meta/about-dialog.vue:9]: Is this change necessary?

File: js/tests/components/layout/main-wrapper/index.spec.js (hunk 1 above)
Comment (phanan) [reply js/tests/components/layout/main-wrapper/index.spec.js:4]: OK let's ignore this for now :)

File: js/components/meta/about-dialog.vue (hunk 2 above)
Comment (Hyzual) [reply js/components/meta/about-dialog.vue:9]: If I leave it with `~#/../img/logo.svg`, webpack errors with the following line:
```
Module not found: Error: Can't resolve 'null' in 'koel/resources/assets/js/components/meta'
```

If I remove the tilde `#/../img/logo.svg`, I get a 404 error :confused: 
It looks like it's necessary, I don't what configuration to change. Where does the tilde `~`  point to by the way ? I have not been able to find out what it means.

Comment (phanan) [reply js/components/meta/about-dialog.vue:9]: Looks like there's a change in how webpack resolves `~`, so now we don't need it anymore. If we replace `~#` with only `#` (or `@` if we prefer, as both `@/../` and `#/../` will resolve into `assets/`), the code will work.

Comment (Hyzual) [reply js/components/meta/about-dialog.vue:9]: OK thanks for the info 🙂. I tried with # but it is interpreted as an html anchor link, so it led to 404.
I'll try with @

Comment (Hyzual) [reply js/components/meta/about-dialog.vue]: It works with `@/../img/logo.svg`

**Comment by phanan** [comment]
All good now, thanks a lot!
